tests/defaults.rs

```diff
@@ -0,0 +1,430 @@
+                .register_taskset(taskset!("About").register_task(task!(get_about)))
+                // Start at least two users, required to run both TaskSets.
+                .set_default(GooseDefault::ThrottleRequests, THROTTLE_REQUESTS)
+                .set_default(GooseDefault::DebugFile, worker_debug_file.as_str())
+                .set_default(GooseDefault::DebugFormat, LOG_FORMAT)
+                .set_default(GooseDefault::MetricsFile, worker_metrics_file.as_str())
+                .set_default(GooseDefault::MetricsFormat, LOG_FORMAT)
+                // Worker configuration using defaults instead of run-time options.
+                .set_default(GooseDefault::Worker, true)
+                .set_default(GooseDefault::ManagerHost, HOST)
+                .set_default(GooseDefault::ManagerPort, PORT)
+                .execute()
+                .unwrap();
+        }));
+    }
+
+    // Start manager instance in current thread and run a distributed load test.
+    let goose_metrics = crate::GooseAttack::initialize_with_config(configuration)
+        .setup()
+        .unwrap()
+        // Alter the name of the task set so NoHashCheck is required for load test to run.
+        .register_taskset(taskset!("FooIndex").register_task(task!(get_index)))
+        .register_taskset(taskset!("About").register_task(task!(get_about)))
+        // Start at least two users, required to run both TaskSets.
+        .set_default(GooseDefault::Host, host.as_str())
+        .set_default(GooseDefault::Users, USERS)
+        .set_default(GooseDefault::RunTime, RUN_TIME)
+        .set_default(GooseDefault::HatchRate, HATCH_RATE)
+        .set_default(GooseDefault::StatusCodes, true)
+        .set_default(GooseDefault::OnlySummary, true)
+        .set_default(GooseDefault::NoTaskMetrics, true)
+        .set_default(GooseDefault::StickyFollow, true)
+        // Manager configuration using defaults instead of run-time options.
+        .set_default(GooseDefault::Manager, true)
+        .set_default(GooseDefault::ExpectWorkers, USERS)
+        .set_default(GooseDefault::NoHashCheck, true)
+        .set_default(GooseDefault::ManagerBindHost, HOST)
+        .set_default(GooseDefault::ManagerBindPort, PORT)
+        .execute()
+        .unwrap();
+
+    // Wait for both worker threads to finish and exit.
+    for worker_handle in worker_handles {
+        let _ = worker_handle.join();
+    }
+
+    let mut metrics_files: Vec<String> = vec![];
+    let mut debug_files: Vec<String> = vec![];
+    for i in 0..USERS {
+        let file = metrics_file.to_string() + &i.to_string();
+        metrics_files.push(file);
+        let file = debug_file.to_string() + &i.to_string();
+        debug_files.push(file);
+    }
+    validate_test(goose_metrics, index, about, &metrics_files, &debug_files);
+}
+
+#[test]
+/// Load test confirming that Goose respects configured defaults.
+fn test_defaults_no_metrics() {
```

Comment:
@LionsAd Originally I had added a `test_defaults_gaggle_no_metrics()` test which is how I realized things don't shut down cleanly in a Gaggle if metrics are disabled. If/when we fix this, where should this test live? Here in `defaults`, or in `gaggle`?